src/store/store.js

```diff
@@ -0,0 +1,28 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+Vue.use(Vuex);
+
+export const store = new Vuex.Store({
+    state: {
+        email: null,
+        failedRequests: [],
+    },
+    actions: {
+        saveEmail(state, email) {
+            return state.commit('saveEmail', email);
```

Comment:
Echale un ojo a esto ->https://vuex.vuejs.org/guide/mutations.html la parte de  **Using Constants for Mutation Types**
Es buena practica definir como constantes los nombres de las mutaciones, y hacerlo usando UPPER_SNAKE_CASE